tools/git-hooks/pre-commit

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+
+# Hide all unstaged changes to prevent any influence of the further checks
+function POP_STASHED() {
+    # pop stashed changes if working directory isn't clean
+    if [ -n "$IS_DIRTY" ]; then
+        git reset --hard HEAD > /dev/null
+        skipped=$(git ls-files -t | grep ^S | cut -d' ' -f2-)
+        git stash pop --index > /dev/null
+        echo "$skipped" | while read file; do git update-index --skip-worktree "$file"; done
+    fi
+}
+
+# determine working tree status
+IS_DIRTY=`[[ $(git diff --shortstat 2> /dev/null | tail -n1) != "" ]] && echo '*'`
+
+# stash not staged for commit changes
+if [ -n "$IS_DIRTY" ]; then
+    git stash save --keep-index
+    trap POP_STASHED EXIT
+fi
+# End of stashing
+
+if git rev-parse --verify HEAD >/dev/null 2>&1
+then
+    against=HEAD
+else
+    # Initial commit: diff against an empty tree object
+    against=4b825dc642cb6eb9a060e54bf8d69288fbee4904
```

Comment:
@LuxoftAKutsan Replaced on initial commit in c7bb3d62f7daa8fbe83ad799c9467954cd7e0912